File: mini-commerce/src/main/java/me/hero/minicommerce/item/application/dto/CreateItemDto.java
package me.hero.minicommerce.item.application.dto;

import lombok.Builder;
import lombok.Value;
import me.hero.minicommerce.item.domain.Item;

@Value
public class CreateItemDto {
  private final Long id;
  private final String name;
  private final Long price;

  @Builder
  public CreateItemDto(Long id, String name, Long price) {
    this.id = id;
    this.name = name;
    this.price = price;
  }

  /**
   * Mapper 를 따로 두지 않고. Item을 의존성으로 가지고 있는 방식
   */
  public CreateItemDto(Item item) {
    this.id = item.getId();
    this.name = item.getName();
    this.price = item.getPrice();
  }
  public Item toEntity() {
    return Item.builder()
        .name(name)
        .price(price)
        .build();
  }
}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/application/port/in/dto/CreateItemDto.kt
package me.hero.minicommerce.item.application.port.`in`.dto

import me.hero.minicommerce.item.domain.Item


data class CreateItemDto(
    val id: Long? = null,
    val name: String,
    val price: Long,
) {
    constructor(item: Item) : this(
        id = item.id,
        name = item.name,
        price = item.price,
    )

    fun toEntity(): Item {
        return Item(
            name = name,
            price = price,
        )
    }
}
